axum-extra/src/either.rs

```diff
@@ -184,17 +219,22 @@ pub enum Either8<E1, E2, E3, E4, E5, E6, E7, E8> {
     E8(E8),
 }
 
+mod private {
+    #[derive(Debug, Clone, Copy)]
+    pub enum EitherOnce {}
+}
+
 macro_rules! impl_traits_for_either {
     (
         $either:ident =>
         [$($ident:ident),* $(,)?],
         $last:ident $(,)?
     ) => {
         #[async_trait]
-        impl<S, B, $($ident),*, $last> FromRequest<S, B> for $either<$($ident),*, $last>
+        impl<S, B, $($ident),*, $last> FromRequest<S, B, private::EitherOnce> for $either<$($ident),*, $last>
         where
             $($ident: FromRequestParts<S>),*,
-            $last: FromRequest<S, B>,
+            $last: FromRequest<S, B, axum_core::extract::private::Once>,
```

Comment:
I wonder how useful this impl is at all, with just the last of the alternatives being allowed to take the request. Maybe we should only provide `FromRequestParts`? We would also need another impl with `TupleOnce` instead `Once` if we wanted to have a request-consuming tuples as `Either`s last type parameter.

Reply:
Yeah I agree. I've removed it.